fedimint-core/src/db/mod.rs

```diff
@@ -286,6 +286,115 @@ pub trait IDatabaseTransaction<'a>: 'a + Send {
     async fn set_tx_savepoint(&mut self);
 }
 
+/// `ISingleUseDatabaseTransaction` re-defines the functions from
+/// `IDatabaseTransaction` but does not consumed `self` when committing to the
+/// database. This allows for wrapper structs to more easily borrow
+/// `ISingleUseDatabaseTransaction` without needing to make additional
+/// allocations.
+#[async_trait]
+pub trait ISingleUseDatabaseTransaction<'a>: 'a + Send {
```

Comment:
why is this `pub`?

Reply:
Crates like `fedimint-rocksdb` use it wrap their normal implementation with this "one time commit"-guard.

Reply:
ah the `fn begin_transaction`